src/e3/main.py

```diff
@@ -144,7 +144,11 @@ def sigterm_handler(sig: int, frame: FrameType | None) -> NoReturn:  # unix-only
             raise SystemExit("SIGTERM received")
 
         if sys.platform != "win32":  # unix-only
-            signal.signal(signal.SIGTERM, sigterm_handler)
+            import threading
```

Comment:
Given that this will be executed by most users of e3, I don't think a huge added value to defer the import of the threading module. Why not moving it to the top of the file?

Reply:
Done.